### PR TITLE
Add a test for algorithm mismatch in X.509 certs

### PR DESCRIPTION
**Pull request by eranmes**

### CODE REVIEW

**Comment by pphaneuf** [comment]
LGTM (once Travis greenlights it!).

**Comment by eranmes** [comment]
PTAL and let me know if you're happy with it.
We now install our own openssl - it's neither pretty nor fast, but it guarantees a good, correct version  (also, no guarantees if somebody's running a log with our open-source code but an older version of openssl)

**Comment by pphaneuf** [comment]
Taking off the LGTM, until I look at it more closely.

File: .travis.yml [hunk 1]
```diff
@@ -20,7 +20,16 @@ before_install:
...
+ - cd openssl-1.0.1l
+ - ./config --prefix=/usr/local --openssldir=/usr/local/openssl
+ - make
+ - sudo make install
+ - cd ..
+
 
 script:
- - make alltests CC=clang CXX=clang++
+ - env OPENSSLDIR=/usr/local/openssl make alltests CC=clang CXX=clang++
```
Comment (pphaneuf) [inline .travis.yml]: Put the ```OPENSSLDIR``` variable at the end, with the others, no need for ```env``` like this...

File: README.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ Auditing for TLS certificates.
 
 ## Dependencies ##
 
- - [OpenSSL](https://www.openssl.org/source/), at least 1.0.0, preferably 1.0.2 (and up)
+ - [OpenSSL](https://www.openssl.org/source/), at least 1.0.0q, preferably 1.0.1l or 1.0.2 (and up)
```
Comment (pphaneuf) [inline README.md]: 1.0.0q has the check we want? Hmm, I'll look into this, there might be a Ubuntu package available somewhere... (but don't wait for that)

**Comment by pphaneuf** [comment]
Not too impressed with making the Travis build time even longer, but oh well...

LGTM

File: README.md (hunk 2 above)
Comment (eranmes) [inline README.md]: Correct, but I've tested with 1.0.1l and there's no reason to point to an even older release.

**Pull request merged**